src/App.js

```diff
@@ -12,24 +14,31 @@ const SORT_BY_LAST_MODIFIED_TIME = ""; //"?sort%5B0%5D%5Bfield%5D=lastModifiedTi
 const AIRTABLE_URL = `${BASE_URL}/${BASE_ID}/${TABLE_NAME}`;
 
 const fetchAirtableData = async ({ method, url, body }) => {
-    const response = await fetch(`${AIRTABLE_URL}${url ?? ""}`, {
-        method,
-        headers: {
-            "Authorization": `Bearer ${API_TOKEN}`,
-            "Content-Type": "application/json"
-        },
-        body: JSON.stringify(body)
-    });
+    const headers = {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${API_TOKEN}`
+    };
     
-    if (!response.ok) {
-        return Promise.reject(`Error: ${response.status}`);
+    let response;
+    switch (method) {
+        case "GET":
+            response = await axios.get(`${AIRTABLE_URL}${url ?? ""}`, { headers });
+            break;
+        case "POST":
+            response = await axios.post(`${AIRTABLE_URL}${url ?? ""}`, body, { headers });
+            break;
+        case "DELETE":
+            response = await axios.delete(`${AIRTABLE_URL}${url ?? ""}`, { headers });
+            break;
+        default:
+            throw new Error("Invalid method type.");
     }
```

Comment:
Nitpick (optimization):
This logic can be simplified using the method ("get", "post", ...etc) as a key. For example, rather than using "GET", you can pass "get" as the method and use it like so (same holds true for the other methods):
```js
const response = await axios[method](...etc);
```
This eliminates the need for a switch/case. If you're worried about passing in an invalid method, then you can do some early check to ensure the method is a part of known/valid key list, otherwise throw an error.

Reply:
Nice! I like that way more!